pkg/querier/queryrange/codec_test.go

```diff
@@ -202,6 +202,57 @@ func Test_codec_EncodeDecodeRequest(t *testing.T) {
 			Step:        30 * 1e3, // step is expected in ms; default is 0 or no step
 			AggregateBy: "series",
 		}, false},
+		{"detected_fields", func() (*http.Request, error) {
+			return DefaultCodec.EncodeRequest(ctx, &DetectedFieldsRequest{
+				logproto.DetectedFieldsRequest{
+					Query:      `{foo="bar"}`,
+					Start:      start,
```

Comment:
Other proto api objects use unix nanos for time, but these 3 (DetectedFields/DetectedLabels/QueryPatterns) use google.protobuf.Timestamp instead. Is there a reason for that? Should we be consistent across all our APIs and use one or the other?